### PR TITLE
[KOGITO-3690] Fix DecisionModelResourcesProvider codegen in case of multiple DMN models

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/KOGITO-3690

From
```java
resourcePaths.add(new org.kie.kogito.dmn.DefaultDecisionModelResource(new org.kie.api.management.GAV("dummy", "dummy", "0.0"), "https://github.com/kiegroup/kogito-examples/dmn-quarkus-listener-example", "LoanEligibility", org.kie.kogito.decision.DecisionModelType.DMN, readResource(org.drools.core.util.IoUtils.class.getClassLoader().getResourceAsStream("LoanEligibility.dmn"))), new org.kie.kogito.dmn.DefaultDecisionModelResource(new org.kie.api.management.GAV("dummy", "dummy", "0.0"), "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF", "Traffic Violation", org.kie.kogito.decision.DecisionModelType.DMN, readResource(org.drools.core.util.IoUtils.class.getClassLoader().getResourceAsStream("Traffic Violation.dmn"))));
resourcePaths.add(new org.kie.kogito.dmn.DefaultDecisionModelResource(new org.kie.api.management.GAV("dummy", "dummy", "0.0"), "https://github.com/kiegroup/kogito-examples/dmn-quarkus-listener-example", "LoanEligibility", org.kie.kogito.decision.DecisionModelType.DMN, readResource(org.drools.core.util.IoUtils.class.getClassLoader().getResourceAsStream("LoanEligibility.dmn"))), new org.kie.kogito.dmn.DefaultDecisionModelResource(new org.kie.api.management.GAV("dummy", "dummy", "0.0"), "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF", "Traffic Violation", org.kie.kogito.decision.DecisionModelType.DMN, readResource(org.drools.core.util.IoUtils.class.getClassLoader().getResourceAsStream("Traffic Violation.dmn"))));
```

To
```java
resourcePaths.add(new org.kie.kogito.dmn.DefaultDecisionModelResource(new org.kie.api.management.GAV("dummy", "dummy", "0.0"), "https://github.com/kiegroup/kogito-examples/dmn-quarkus-listener-example", "LoanEligibility", org.kie.kogito.decision.DecisionModelType.DMN, readResource(org.drools.core.util.IoUtils.class.getClassLoader().getResourceAsStream("LoanEligibility.dmn"))));
resourcePaths.add(new org.kie.kogito.dmn.DefaultDecisionModelResource(new org.kie.api.management.GAV("dummy", "dummy", "0.0"), "https://github.com/kiegroup/drools/kie-dmn/_A4BCA8B8-CF08-433F-93B2-A2598F19ECFF", "Traffic Violation", org.kie.kogito.decision.DecisionModelType.DMN, readResource(org.drools.core.util.IoUtils.class.getClassLoader().getResourceAsStream("Traffic Violation.dmn"))));
```

### CODE REVIEW

**Review by r00ta — APPROVED** [review]
Thank you very much for the fix!

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/decision/DecisionModelResourcesProviderGeneratorTest.java [hunk 1]
```diff
@@ -95,5 +108,12 @@ public void generateDecisionModelResourcesProvider() throws Exception {
 
         final NameExpr nameExpr = call.getScope().get().asNameExpr();
         assertEquals(nameExpr.getName().getIdentifier(), "resourcePaths");
+
+        long numberOfAddStms = body.getStatements().stream()
```
Comment (manstis) [inline kogito-codegen/src/test/java/org/kie/kogito/codegen/decision/DecisionModelResourcesProviderGeneratorTest.java:112]: Is it worth checking the `add` statements has different `DefaultDecisionModelResource` instances?

The old, broken way, had two _adds_ but they were for the same instance.

Comment (danielezonca) [reply kogito-codegen/src/test/java/org/kie/kogito/codegen/decision/DecisionModelResourcesProviderGeneratorTest.java:117]: Good point, done with https://github.com/kiegroup/kogito-runtimes/pull/847/commits/496d42bb9e6b5495dae2546b8e1d9b8e9c929243

Comment (manstis) [reply kogito-codegen/src/test/java/org/kie/kogito/codegen/decision/DecisionModelResourcesProviderGeneratorTest.java:117]: Sweet.

**Review by manstis — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**